### DISCUSSION TITLE
Changes aka. change-log.

### DISCUSSION THREAD

**Discussion by bert269** [toverainc/willow#149] [Ideas]
Is there a posibility that we can get a list of changes to the code and when it was made?
This way, we can decide which code updates are important/applicble and decide to update or not.

I looked around on Github project page, but could not find a changelog like this - did I miss it?

**Comment by kristiankielhofner** [comment]
We will make changelogs when we do official Github releases. For the time being we strongly encourage everyone to ```git pull``` main as often as is convenient.

With releases we will have the ability to manage versions, do over the air updates, etc.

**Comment by bert269** [comment]
Thanks for the update - I totally understand.

I have a question on the 'git pull' sequence. After I do that, do I
continue the full 'utils.sh docker-build', '-docker', '-install', '-config'
and '-build' each time, or do I skip some of the steps?
I asked, because once I am in docker and want to redo the install - it tells
me that the 'you alrady have a deps directory -exiting' and there is no way for me
around this other than to start from scratch with another 'git clone' as
before.(which will now be a 'git pull')

Thank you

**Reply by kristiankielhofner** [reply]
For nearly everything in recent memory you can just ```git pull``` and ```./utils.sh build```. If you want to peek at any new config options or features you can run ```./utils.sh config``` but it's not necessary.

**Reply by tombloemker** [reply]
Something is wonky in my configuration menu, is there a way to reset it to the original template?

**Reply by hamishcunningham** [reply]
Move sdkconfig somewhere and rerun config should do it I think

**Reply by tombloemker** [reply]
A search generated the following sdkconfig files, 

sdkconfig
sdkconfig.defaults.esp32s2
sdkconfig.defaults
sdkconfig.h

Just wanted to make sure you meant "sdkconfig" before I blow up my whole project.

**Reply by nikito** [reply]
What branch are you on? A lot of this config stuff is going to change in 1.0 and be done mainly on WAS.

**Reply by tombloemker** [reply]
Okay, really it was an excuse to learn something new. I guess I'll wait for 1.0. I hate to press the easy button but I've pestered yall too much already.

**Comment by stintel** [comment]
Until we get to releases, I suggest looking at [closed pull requests](https://github.com/toverainc/willow/pulls?q=is%3Apr+is%3Aclosed) for an overview of changes.

**Comment by hamishcunningham** [comment]
hmmm... maybe I'm out of context; no sdkconfig.willow? on the main branch
this is copied to sdkconfig by the install or reinstall steps, then
"utils.sh config" (called from within the docker container) will allow you
to alter the settings

your sdkconfigs look different tho...

On Thu, 13 Jul 2023 at 17:04, tombloemker ***@***.***> wrote:

> A search generated the following sdkconfig files,
>
> sdkconfig
> sdkconfig.defaults.esp32s2
> sdkconfig.defaults
> sdkconfig.h
>
> Just wanted to make sure you meant "sdkconfig" before I blow up my whole
> project.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/toverainc/willow/discussions/149#discussioncomment-6441053>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBRBFCPZ5XC3BJLDNKYA6LXQAMB7ANCNFSM6AAAAAAY3DLRYE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 

*Hamish Cunningham*
Professor of Computer Science, University of Sheffield, UK


*I used to hope that as time passed I would get older and wiser,
        but it seems that in fact I just get odder and wider.*

*Phone: 07740 149 842 / 0114 222 1891             * *unphone.net*
<https://unphone.net/>
***@***.***            ***@***.***
https://hamish.gate.ac.uk    *https://iot.unphone.net
<https://iot.unphone.net/>*
https://gate.ac.uk                 https://pi.gate.ac.uk

*https://www.sheffield.ac.uk/dcs/people/academic/hcunningham
<https://www.sheffield.ac.uk/dcs/people/academic/hcunningham>*
